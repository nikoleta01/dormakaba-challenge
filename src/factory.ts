import { ConnectionStatus } from './models/ConnectionStatus';
import { ConnectionType } from './models/ConnectionType';
import { Door } from './models/Door';
import { faker } from '@faker-js/faker';

export function buildDoor(door?: Partial<Door>) {
  return {
    id: faker.string.uuid(),
    name: faker.lorem.word(),
    buildingName: faker.lorem.words(2),
    connectionType: faker.helpers.enumValue(ConnectionType),
    connectionStatus: faker.helpers.enumValue(ConnectionStatus),
    lastConnectionStatusUpdate: faker.date.past().toISOString(),
    apartmentName: `Apartment ${faker.number.int(1000)}`,
    ...door,
  };
}
