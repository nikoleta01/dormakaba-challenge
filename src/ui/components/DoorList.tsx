import { useRouter } from 'next/router';
import { useCallback } from 'react';
import { Door } from '@/models/Door';
import { DataGrid, GridColDef, GridRowParams } from '@mui/x-data-grid';
import Typography from '@mui/material/Typography';

interface DoorListProps {
  doors: Door[];
}

const connectionColor = (door: Door) =>
  door.connectionStatus === 'online' ? 'success.main' : 'error.main';

const columns: GridColDef<Door>[] = [
  {
    field: 'name',
    headerName: 'Name',
    flex: 1,
  },
  {
    field: 'buildingName',
    headerName: 'Building',
    flex: 1,
  },
  {
    field: 'connectionType',
    headerName: 'Connection type',
    flex: 1,
  },
  {
    field: 'connectionStatus',
    headerName: 'Connection status',
    flex: 1,
    renderCell: ({ row: door }) => {
      return (
        <Typography component="span" color={connectionColor(door)}>
          {door.connectionStatus}
        </Typography>
      );
    },
  },
  {
    field: 'apartmentName',
    headerName: 'Apartment name',
    flex: 1,
    renderCell: ({ row: door }) => {
      return <Typography component="span">{door.apartmentName}</Typography>;
    },
  },
];

export function DoorList({ doors }: DoorListProps) {
  const router = useRouter();

  const onDoorRowClick = useCallback(
    (gridRow: GridRowParams<Door>) => {
      router.push({
        pathname: '/doors/[doorId]',
        query: { doorId: gridRow.id },
      });
    },
    [router],
  );

  return (
    <DataGrid
      hideFooter
      rows={doors}
      columns={columns}
      disableRowSelectionOnClick
      onRowClick={onDoorRowClick}
    />
  );
}
