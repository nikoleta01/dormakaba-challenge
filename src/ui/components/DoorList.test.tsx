import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Door } from '@/models/Door';
import { DoorList } from './DoorList';
import { buildDoor } from '@/factory';
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({ query: {} });

function renderDoorList(doors: Door[]) {
  return render(<DoorList doors={doors} />);
}

describe('DoorList', () => {
  it('redirects to door detail on row click', () => {
    const mockPush = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });

    const door1 = buildDoor({
      id: '1',
      name: 'New Building',
    });
    const door2 = buildDoor({
      id: '2',
      name: 'Old Building',
    });

    renderDoorList([door1, door2]);

    const gridCell = screen.getByRole('gridcell', { name: 'New Building' });
    fireEvent.click(gridCell);

    expect(mockPush).toHaveBeenCalledWith({
      pathname: '/doors/[doorId]',
      query: { doorId: '1' },
    });
  });
  it('renders the door list with proper data', () => {
    const door1 = buildDoor({
      name: 'Front Door',
      buildingName: 'Building A',
      connectionType: 'wired',
      connectionStatus: 'online',
      apartmentName: 'Apartment 1A',
    });
    const door2 = buildDoor({
      name: 'Back Door',
      buildingName: 'Building B',
      connectionType: 'wireless',
      connectionStatus: 'offline',
      apartmentName: 'Apartment 2B',
    });
    renderDoorList([door1, door2]);

    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.getByText('Building')).toBeInTheDocument();
    expect(screen.getByText('Connection type')).toBeInTheDocument();
    expect(screen.getByText('Connection status')).toBeInTheDocument();
    expect(screen.getByText('Apartment name')).toBeInTheDocument();

    expect(screen.getByText('Front Door')).toBeInTheDocument();
    expect(screen.getByText('Building A')).toBeInTheDocument();
    expect(screen.getByText('wired')).toBeInTheDocument();
    expect(screen.getByText('online')).toBeInTheDocument();
    expect(screen.getByText('Apartment 1A')).toBeInTheDocument();

    expect(screen.getByText('Back Door')).toBeInTheDocument();
    expect(screen.getByText('Building B')).toBeInTheDocument();
    expect(screen.getByText('wireless')).toBeInTheDocument();
    expect(screen.getByText('offline')).toBeInTheDocument();
    expect(screen.getByText('Apartment 2B')).toBeInTheDocument();
  });
  it('renders connection status with correct color', () => {
    const door1 = buildDoor({
      connectionStatus: 'online',
    });
    const door2 = buildDoor({
      connectionStatus: 'offline',
    });
    renderDoorList([door1, door2]);

    const onlineStatus = screen.getByText('online');
    const offlineStatus = screen.getByText('offline');

    expect(onlineStatus).toHaveStyle('color: rgb(46, 125, 50)');
    expect(offlineStatus).toHaveStyle('color: rgb(211, 47, 47)');
  });
});
