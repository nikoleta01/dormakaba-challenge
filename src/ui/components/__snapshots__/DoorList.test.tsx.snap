// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoorList should render correctly 1`] = `
<div
  class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-root--noToolbar MuiDataGrid-withBorderColor css-5wdp45-MuiDataGrid-root"
  style="--DataGrid-width: 0px; --DataGrid-hasScrollX: 0; --DataGrid-hasScrollY: 1; --DataGrid-scrollbarSize: 0px; --DataGrid-rowWidth: 0px; --DataGrid-columnsTotalWidth: 0px; --DataGrid-leftPinnedWidth: 0px; --DataGrid-rightPinnedWidth: 0px; --DataGrid-headerHeight: 56px; --DataGrid-headersTotalHeight: 56px; --DataGrid-topContainerHeight: 56px; --DataGrid-bottomContainerHeight: 0px; --height: 52px;"
>
  <div
    aria-colcount="4"
    aria-multiselectable="false"
    aria-rowcount="2"
    class="MuiDataGrid-main css-1bgckiu-MuiDataGrid-main"
    role="grid"
    tabindex="-1"
  >
    <div
      class="css-18l11qw"
      data-id="gridPanelAnchor"
      role="presentation"
    />
    <div
      class="MuiDataGrid-virtualScroller css-1vigakz-MuiDataGrid-virtualScroller"
      role="presentation"
      style="overflow-x: hidden;"
    >
      <div
        class="MuiDataGrid-topContainer MuiDataGrid-container--top css-9dbfce"
        role="presentation"
      >
        <div
          class="MuiDataGrid-columnHeaders css-1gtv474-MuiDataGrid-columnHeaders"
          role="presentation"
        >
          <div
            aria-rowindex="1"
            class="MuiDataGrid-row--borderBottom css-yseucu-MuiDataGrid-columnHeaderRow"
            role="row"
          >
            <div
              role="presentation"
              style="width: 0px;"
            />
            <div
              aria-colindex="1"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
              data-field="name"
              role="columnheader"
              style="height: 56px; width: 0px;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
                role="presentation"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                  role="presentation"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Name
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                      field="name"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    id=":r1:"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 56px;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <rect
                    height="24"
                    rx="0.5"
                    width="1"
                    x="11.5"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
              data-field="buildingName"
              role="columnheader"
              style="height: 56px; width: 0px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
                role="presentation"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                  role="presentation"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Building
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                      field="buildingName"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    id=":r7:"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 56px;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <rect
                    height="24"
                    rx="0.5"
                    width="1"
                    x="11.5"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="3"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor"
              data-field="connectionType"
              role="columnheader"
              style="height: 56px; width: 0px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
                role="presentation"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                  role="presentation"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Connection type
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                      field="connectionType"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    id=":rd:"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 56px;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <rect
                    height="24"
                    rx="0.5"
                    width="1"
                    x="11.5"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="4"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable MuiDataGrid-withBorderColor MuiDataGrid-columnHeader--lastUnpinned MuiDataGrid-columnHeader--last"
              data-field="connectionStatus"
              role="columnheader"
              style="height: 56px; width: 0px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
                role="presentation"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                  role="presentation"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitle css-1gqmilo-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Connection status
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                      field="connectionStatus"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-120dh41-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-r363y4-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    id=":rj:"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--resizable MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 56px;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <rect
                    height="24"
                    rx="0.5"
                    width="1"
                    x="11.5"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiDataGrid-filler"
              role="presentation"
            />
            <div
              class="MuiDataGrid-scrollbarFiller MuiDataGrid-scrollbarFiller--header"
              role="presentation"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiDataGrid-virtualScrollerContent css-aymtem-MuiDataGrid-virtualScrollerContent"
        role="presentation"
        style="width: auto; flex-shrink: 0; flex-basis: 52px;"
      >
        <div
          class="MuiDataGrid-virtualScrollerRenderZone css-11dqcl8-MuiDataGrid-virtualScrollerRenderZone"
          role="rowgroup"
          style="transform: translate3d(0, 0px, 0);"
        >
          <div
            aria-rowindex="2"
            aria-selected="false"
            class="MuiDataGrid-row MuiDataGrid-row--firstVisible MuiDataGrid-row--lastVisible"
            data-id="63f637c9f3c48a124616044b"
            data-rowindex="0"
            role="row"
            style="max-height: 52px; min-height: 52px; --height: 52px;"
          >
            <div
              class="MuiDataGrid-cellOffsetLeft"
              role="presentation"
              style="width: 0px;"
            />
            <div
              aria-colindex="1"
              aria-colspan="1"
              aria-rowspan="1"
              class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
              data-colindex="0"
              data-field="name"
              role="gridcell"
              style="--width: 0px;"
              tabindex="-1"
              title="Building Main Entrance"
            >
              Building Main Entrance
            </div>
            <div
              aria-colindex="2"
              aria-colspan="1"
              aria-rowspan="1"
              class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
              data-colindex="1"
              data-field="buildingName"
              role="gridcell"
              style="--width: 0px;"
              tabindex="-1"
              title="Bahnhofstrasse 10A"
            >
              Bahnhofstrasse 10A
            </div>
            <div
              aria-colindex="3"
              aria-colspan="1"
              aria-rowspan="1"
              class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
              data-colindex="2"
              data-field="connectionType"
              role="gridcell"
              style="--width: 0px;"
              tabindex="-1"
              title="wired"
            >
              wired
            </div>
            <div
              aria-colindex="4"
              aria-colspan="1"
              aria-rowspan="1"
              class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
              data-colindex="3"
              data-field="connectionStatus"
              role="gridcell"
              style="--width: 0px;"
              tabindex="-1"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 css-wl91uy-MuiTypography-root"
              >
                online
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiDataGrid-bottomContainer MuiDataGrid-container--bottom css-kkb6k7"
        role="presentation"
      />
    </div>
    <div
      aria-hidden="true"
      class="MuiDataGrid-scrollbar MuiDataGrid-scrollbar--vertical css-20a7a0"
      tabindex="-1"
    >
      <div
        class="MuiDataGrid-scrollbarContent"
      />
    </div>
  </div>
</div>
`;
